glean-core/ffi/src/upload.rs

```diff
@@ -147,7 +145,7 @@ impl Drop for FfiPingUploadTask {
             unsafe {
                 glean_str_free(*document_id);
                 glean_str_free(*path);
-                glean_str_free(*body);
+                //body.destroy(); // TODO FIXME: glean_str_free(*body);
```

Comment:
@badboy this is troubling me a bit - I don't seem to find a way to call `body.destroy()`, because `body` has been moved already. Clues?

Reply:
Ah damn, yes.
I _think_ what we need to do is along the following lines ([full example with vec instead of ByteBuffer](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=dc33abca36e24dfb7543881047dbf582))

```rust
let body = mem::replace(body, ByteBuffer::new_with_size(0));
body.destroy();
``` 

What happens here is we take ownership of the `body`, but we need to put something in its place, so we just use an empty byte buffer (there needs to be _something_ because after this `drop` function runs, Rust will clean out all remaining stuff of memory and it can't know we already did that in parts). A byte buffer with size `0` creates a Vec with size 0, which does not allocate, so we don't create any new memory we would need to drop.

An alternative would be to have the `body` be an [`ManuallyDrop<ByteBuffer>`](https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html), so we can take it out. That's essentially the same (but will involve an `unsafe` call).